src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -805,9 +805,11 @@ protected void forceShowWindowIfHidden() {
 		) {
 			requestShowSelf(InputMethodManager.SHOW_IMPLICIT);
 		}
+		else{
+			super.showWindow(true);
+		}
```

Comment:
Shouldn't this be:
```suggestion
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
			requestShowSelf(InputMethodManager.SHOW_IMPLICIT);
		} else {
			super.showWindow(true);
		}
```
?

The Main View must definitely not be displayed in Mode Passthrough. That's the one used when dialling a number or other scenarios when the keyboard must remain invisible.